### ISSUE TITLE
Add language in same spot on scraper show page as it is on scraper index page

### ISSUE THREAD

**Issue by mlandauer** [openaustralia/morph#664]
For consistency. It currently is on the sidebar in the "code" section. Should it remain there as well or should it just be at the top?

**Comment by mlandauer** [comment]
I'll wait until your fix for this is merged and then I'll deploy to production

**Comment by equivalentideas** [comment]
After putting this in, I'm not sure about it. I don't think there is a great need for consistency here as the page is so different. Is this information so important on this page? To me it feels more useful on the index page where you're making comparisons/searching—but it's not so useful here to justify being pushed so prominently. What do you think @mlandauer ?

![screen shot 2015-04-29 at 4 11 31 pm](https://cloud.githubusercontent.com/assets/1239550/7385481/7296d468-ee8a-11e4-9f30-1fe68fc8919c.png)

It feels like it wants to be part of a 'scraper info' section or something, like the tops of github repos:

![screen shot 2015-04-29 at 4 15 01 pm](https://cloud.githubusercontent.com/assets/1239550/7385509/edc2e140-ee8a-11e4-89f1-9dadfd94c088.png)

**Comment by mlandauer** [comment]
Hmmm... I see what you mean. The main bit of the scraper page is really focused on the running of the scraper and the data. It really doesn't matter what language it is. The bit where that is is that sidebar.

Perhaps, then would there be a way that you could change the way the language is displayed in the "Scraper code" sidebar so that it mirrors the way it's displayed on the index pages.

Also, if people are looking for the word "Ruby", for instance on the page because that's what their interested in and they easily find it in the "Scraper code" section because it's displayed in a similar way then they find the code bit which is what they're interested in. So, I think that would work.

**Comment by henare** [comment]
I just mentioned to @equivalentideas a random idea.

What about instead of logos for languages, as was mentioned elsewhere, we use Bootstrap labels similar to the "errored" one we already use. We then add a very muted background colour that represents the language.

This may help this issue because them the language looks less like another bit of text.

**Comment by mlandauer** [comment]
@henare that's what I was imagining when @equivalentideas suggested colours and I suggested low-saturation colours.

**Comment by equivalentideas** [comment]
> Perhaps, then would there be a way that you could change the way the language is displayed in the "Scraper code" sidebar so that it mirrors the way it's displayed on the index pages.

I think that comes together really nicely with what @henare is saying. We could add this slight tab effect on the indexes and then use it in the side bar there—with a touch of the desaturated colour.

:+1:

**Comment by equivalentideas** [comment]
So what should we do for the colours? We could copy githubs colours for different languages they use in that bar up the top of repos.

**Comment by mlandauer** [comment]
Can we move the colour thing to a new ticket and do that separately? I'll create a ticket

**Comment by equivalentideas** [comment]
good idea.

**Comment by equivalentideas** [comment]
I'm going to close this as I think we decided to go with a different approach, which is also pending on #688 which has been pushed back. Please reopen if you disagree.

**Comment by mlandauer** [comment]
There isn't another ticket addressing this so I'm reopening

**Comment by mlandauer** [comment]
I'm finding this a bit frustrating. There is specific feedback here which has been ignored
> Perhaps, then would there be a way that you could change the way the language is displayed in the "Scraper code" sidebar so that it mirrors the way it's displayed on the index pages.

> Also, if people are looking for the word "Ruby", for instance on the page because that's what their interested in and they easily find it in the "Scraper code" section because it's displayed in a similar way then they find the code bit which is what they're interested in. So, I think that would work.

**Comment by equivalentideas** [comment]
I'm sorry you feel that you're being ignored, I just missed that in all the discussion above.

> Perhaps, then would there be a way that you could change the way the language is displayed in the "Scraper code" sidebar so that it mirrors the way it's displayed on the index pages.

> Also, if people are looking for the word "Ruby", for instance on the page because that's what their interested in and they easily find it in the "Scraper code" section because it's displayed in a similar way then they find the code bit which is what they're interested in. So, I think that would work.

I think that's a good summary of what this issue is about now, better than the title (I think that's why I missed this). Thanks for describing the use case. I agree, the scraper language information should have a consistent style so it's easily recognisable.

Here's a quick example of what that could look like using the current display of language in the index:

![screen shot 2015-04-30 at 3 58 40 pm](https://cloud.githubusercontent.com/assets/1239550/7407141/d3120098-ef51-11e4-9855-95e03ab3670f.png)

**Comment by mlandauer** [comment]
Thanks for putting together an example. The hierarchy here seems too jumbled.

Here's an idea to throw into the mix. What if the scraper code section looked something like this

# Scraper code
Ruby
[morphio-congressmen-helper](https://github.com/congresointeractivo/morphio-congressmen-helper)/[scraper.rb](https://github.com/congresointeractivo/morphio-congressmen-helper/blob/master/scraper.rb)

git clone URL
...

where the repo part of the path links to the main repo and the scraper.rb part of the path links to just the scraper code. Gives the functionality of what was there before without the complexity and also makes the relationship between the repo and the file a little clearer by using file paths (which are familiar to a coder) and used elsewhere in the site

**Comment by equivalentideas** [comment]
> Gives the functionality of what was there before without the complexity and also makes the relationship between the repo and the file a little clearer by using file paths (which are familiar to a coder) and used elsewhere in the site

I quite like that @mlandauer . I also like that it simplifies things.

On wide screens (majority of users) it will almost always wrap over a line or two even. I don't think that's too bad, and we can always iterate.

Should I do a PR of this?

**Comment by mlandauer** [comment]
Cool. PR would be great!

Yeah the line wrapping is an issue but not a huge one. That sidebar has always felt a bit squashed to me but that's definitely something to think about another time :-)

**Comment by equivalentideas** [comment]
on this now

**Comment by mlandauer** [comment]
Closed by #684
